src/utils/buffer.cpp

```diff
@@ -38,10 +39,11 @@ Buffer::Buffer(struct iio_device *dev) {
 		m_dev = nullptr;
 		THROW_M2K_EXCEPTION("Buffer: Device not found, so no buffer can be created", libm2k::EXC_INVALID_PARAMETER);
 	}
+	m_dev_name = iio_device_get_name(m_dev);
```

Comment:
can this be placed in the initialiser list? I assume m_dev is know and it is the iio_device passed in Buffer(..)

Reply:
I thinks is a better approach to access the device structure after validating its integrity.